src/app/components/ToolBar.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import React, { useState } from 'react';
+import { ShotsTypes, userInputsArr } from '../../store/inputStore';
+
+interface toolBarInter {
+    isLoading: boolean;
+    send: (examples: ShotsTypes[], userInputs: string) => Promise<void>;
+}
+
+export const ToolBar = ({ isLoading, send }: toolBarInter) => {
+    const [
+        updateUserInputArr,
+        userInput,
+        updateUserInput,
+        userResults,
+        updateUserResults,
+        examples,
+        deleteUserInputArr,
+        loadDefaultExample,
+    ] = userInputsArr((state) => [
+        state.updateUserInputArr,
+        state.userInputs,
+        state.updateUserInput,
+        state.userResults,
+        state.updateUserResults,
+        state.examples,
+        state.deleteUserInputArr,
+        state.loadDefaultExample,
+    ]);
+
+    const [mode, useMode] = useState(true);
+
+    const buttonClassName = isLoading ? ' text-gray-400' : ' hover:bg-slate-600';
+
+    const disabledSendRequest = !(examples.length >= 1 && userInput !== '' && !isLoading);
+    const classNameForSend = disabledSendRequest ? ' text-gray-400' : ' hover:bg-yellow-800';
+    const classNameForButton = isLoading ? ' text-gray-400' : ' hover:bg-red-900';
+
+    const clickedAdd = () => {
+        if (userInput.trim() !== '' && userResults.trim() !== '') {
+            updateUserInputArr({ input: userInput, result: userResults });
+            updateUserInput('');
+            updateUserResults('');
+        }
+    };
+
+    const handleDeleteAll = () => {
+        deleteUserInputArr();
+        updateUserInput('');
+        updateUserResults('');
+    };
+
+    const onClickHandler = async () => {
+        await send(examples, userInput);
+    };
+
+    const useTheMode = (bol: boolean) => {
```

Comment:
Rename and avoid using useXXX for methods that aren't hooks. You are triggering some checks because of this.